added button and single input components

jrblaettler opened the pull request:
Added the two requested components. The current input component is only able to do a single input so might just factor it out to be Form and FormInput but we have a single use input now.
Need to add the ability to add CSS to the input but Im gonna do that when I add tailwind and looking at styling in a bigger way.

jrblaettler committed:
Changes committed:

	new file:   components/Button.tsx
	new file:   components/Input.tsx
	new file:   test/components/Button.test.tsx
	new file:   test/components/Input.test.tsx
	renamed:    test/index.test.tsx -> test/pages/index.test.tsx

jrblaettler committed:
Changes committed:

	modified:   components/Input.tsx
	modified:   test/components/Input.test.tsx

jrblaettler committed:
Changes committed:

	modified:   components/Button.tsx

jrblaettler committed:
Changes committed:

	modified:   components/Button.tsx
	modified:   components/Input.tsx
	modified:   test/components/Button.test.tsx
	modified:   test/components/Input.test.tsx

jochamb commented:
i realize i forgot to mention this, but please copy over the tasks i made in the other repository and create a "kanban board" project in this repo -> you can then link pull requests to those tasks and have them auto close as you merge the pull requests

jrblaettler committed:
Changes committed:

	modified:   components/Input.tsx

jochamb commented on components/Button.tsx:
```diff
@@ -0,0 +1,17 @@
+}
+const Button = (props: ButtonProps) => {
+  return (
+    <button className={props.className} onClick={() => props.onClick()} disabled={props.disabled}>
```
```typescript
onClick={() => props.onClick()}
```

can be simplified to
```typescript
onClick={props.onClick}
```

jochamb replied:
this wasnt changed?

edit: nvm it was, didnt see out of date tag

jrblaettler committed:
Changes committed:

	modified:   components/Input.tsx

jochamb commented on components/Input.tsx:
```diff
@@ -0,0 +1,58 @@
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
```
i wouldnt have a form element be a part of the `Input` component. even if its only temporary. either remove the `form` from the input or create a resuable `Form` in this pull request and update

jrblaettler commented:
Will do.

jochamb commented on components/Input.tsx:
```diff
@@ -0,0 +1,58 @@
+      <input
+        id={props.id}
+        name={props.name}
+        type={props.type || 'text'}
```
the default input type is `text`, so you could omit it if you wanted

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types

jochamb commented on test/components/Button.test.tsx:
```diff
@@ -0,0 +1,24 @@
+import Button from '../../components/Button';
```
small nit: prefer to stick to module imports instead of using relative

jochamb replied:
nextjs i believe handles this automatically, so can just do `components/Button`

jochamb commented on test/components/Input.test.tsx:
```diff
@@ -0,0 +1,43 @@
+import Input from '../../components/Input';
```
relative import

jochamb replied:
if youre struggling to get it working in jest

https://jestjs.io/docs/configuration#modulenamemapper-objectstring-string--arraystring

```javascript
{
  moduleNameMapper: {
    '^/components(.*)$': '<rootDir>/components$1'
    '^/pages(.*)$': '<rootDir>/pages$1'
  },
}
etc

jochamb commented on test/components/Input.test.tsx:
```diff
@@ -0,0 +1,43 @@
+import Input from '../../components/Input';
+import { screen, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import Button from '../../components/Button';
```
relative import

jrblaettler committed:
Changes committed:

	modified:   components/Button.tsx
	modified:   test/components/Button.test.tsx
	modified:   test/components/Input.test.tsx
	modified:   tsconfig.json

jrblaettler committed:
Changes committed:

	modified:   components/Button.tsx

jrblaettler committed:
Changes committed:

	modified:   components/Input.tsx

jrblaettler committed:
Changes committed:

	modified:   components/Input.tsx
	modified:   test/components/Input.test.tsx

jrblaettler commented:
Made the suggested changes

jrblaettler committed:
Changes committed:

	modified:   test/components/Input.test.tsx
